### PR TITLE
Trim platfrom libraries from deps file

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: src/dotnet/commands/dotnet-publish/PublishCommand.cs [hunk 1]
```diff
@@ -173,7 +173,10 @@ private bool PublishProjectContext(ProjectContext context, string buildBasePath,
             {
                 // Make executable in the new location
                 var executable = new Executable(context, buildOutputPaths, outputPath, buildOutputPaths.IntermediateOutputDirectoryPath, exporter, configuration);
-                executable.WriteConfigurationFiles(exports, FilterExports(exports, buildExclusionList), includeDevConfig: false);
+                var runtimeExports = exports.FilterExports(allExclusionList);
```
Comment (eerhardt) [inline src/dotnet/commands/dotnet-publish/PublishCommand.cs]: Can this result be cached above so we don't need to call this method twice?

File: src/Microsoft.DotNet.Compiler.Common/Executable.cs [hunk 2]
```diff
@@ -159,7 +159,12 @@ private void WriteDepsFileAndCopyProjectDependencies(LibraryExporter exporter)
         {
             // When called this way we don't need to filter exports, so we pass the same list to both.
```
Comment (eerhardt) [inline src/Microsoft.DotNet.Compiler.Common/Executable.cs]: comment seems out of date now.

**Comment by eerhardt** [comment]
looks good.  just a couple tiny comments

**Comment by eerhardt** [comment]
:shipit: 

Fixes #3519

**Comment by schellap** [comment]
@pakrym, confirming that build and publish use consistent code paths, correct? i.e., build=copy to deps file. publish=copy to file system.

Also wanted to confirm if native and resources were trimmed as well. Thanks!

**Comment by LonghronShen** [comment]
Hi there, I'm using .NET Core 1.0 RTM, and the trimming problem is still there.
As far as I see, AutoMapper 5.1.1 and some portable class libraries won't load correctly. It says:

`Error: assembly specified in the dependencies manifest was not found -- package: 'AutoMapper', version: '5.1.1', path: 'lib/netstandard1.3/AutoMapper.dll'` 

or

'Error: assembly specified in the dependencies manifest was not found -- package: 'TestPortableClassLibrary', version: '1.0', path: 'lib/portable45-net45+win8+wpa81/CloOpenSDK.dll''

Any idea about this? Thank you!

**Comment by piotrpMSFT** [comment]
@longhornshen can you open an issue and link this pr?

**Comment by piotrpMSFT** [comment]
Also, a repro project would be wonderful!

**Comment by LonghronShen** [comment]
Yeah, OK, I'll try to make a demo repo to help reproduce the problem and then create a new issue linking to this pr.

**Comment by piotrpMSFT** [comment]
Thanks!

**Pull request merged**